cmd/rpcdaemon/commands/eth_block.go

```diff
@@ -314,19 +314,19 @@ func (api *APIImpl) GetBlockTransactionCountByNumber(ctx context.Context, blockN
 		n := hexutil.Uint(len(b.Transactions()))
 		return &n, nil
 	}
-	blockNum, _, _, err := rpchelper.GetBlockNumber(rpc.BlockNumberOrHashWithNumber(blockNr), tx, api.filters)
+	blockNum, blockHash, _, err := rpchelper.GetBlockNumber(rpc.BlockNumberOrHashWithNumber(blockNr), tx, api.filters)
 	if err != nil {
 		return nil, err
 	}
-	body, _, txAmount, err := rawdb.ReadBodyByNumber(tx, blockNum)
+	body, err := api._blockReader.Body(ctx, tx, blockHash, blockNum)
 	if err != nil {
 		return nil, err
 	}
 	if body == nil {
 		return nil, nil
 	}
-	n := hexutil.Uint(txAmount)
-	return &n, nil
+	txAmount := hexutil.Uint(len(body.Transactions) - 2) // 1 system txn in the begining of block, and 1 at the end
```

Comment:
I don't think `-2` is needed. Did you do a test for this?

Reply:
I haven't don't a test for this but I followed `rawdb.ReadBody` which subtracted two. I'll add a test for this method.

Reply:
Only db has system transactions, but not block object

Reply:
yeah also `api._blockReader.Body()` method didn't return the block body with the transactions...